### PR TITLE
[OCPCLOUD-1066] Makefile fix 

### PR DESCRIPTION
**Pull request by SamuelStuchly**
Fixed broken Makefile targets.
Added script to fix permission issues.
Updated Readme.
Update .gitignore.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,13 +16,27 @@ GOPROXY ?=
...
+	ENGINE=podman
+else ifeq ($(shell command -v docker > /dev/null 2>&1 ; echo $$? ), 0)
+	ENGINE=docker
+else
+	NO_DOCKER=1
+endif
+
+USE_DOCKER ?= 0
+ifeq ($(USE_DOCKER), 1)
+	ENGINE=docker
```
Comment (elmiko) [inline Makefile:30]: extra spaces after `docker`

**Review by elmiko — APPROVED** [review]
tested out locally, worked as expected. one minor nit
/approve

**Review by elmiko — APPROVED** [review]
/lgtm

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,7 @@
 
 bin/
 .gobincache/
+
+# IDE configs
+.idea/
+.vscode/
```
Comment (JoelSpeed) [inline .gitignore:19]: It's convention to leave a blank new line at the end of every file, please make sure your IDE is set up to do this (this is the red marker on this line that shows it's missing)

File: hack/owner_reset.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+sudo chown -R $USER bin \
+    chown $USER .git/index
+
```
Comment (JoelSpeed) [inline hack/owner_reset.sh:5]: Whitespace on this line is also confusing it
```suggestion
```

**Review by JoelSpeed — COMMENTED** [review]
Couple of nits, otherwise this LGTM

**Pull request merged**